sklearn/preprocessing/data.py

```diff
@@ -1325,6 +1325,21 @@ def normalize(X, norm='l2', axis=1, copy=True, return_norm=False):
     return_norm : boolean, default False
         whether to return the computed norms
 
+    Returns
+    -------
+    X : {array-like, sparse matrix}, shape [n_samples, n_features]
+        Normalized input X.
+
+    norms : array
+        When X is sparse, an NotImplementedError will be raised for
+        norm 'l1' or 'l2'; an array with same shape with X.data,
```

Comment:
That's definitely a reasonable way to do it. 
Actually after thinking this through, I found a good reason:
I use `normalize()`to normalize along a specific dimension. Even if the matrix is sparse, I don't think there will be a **whole** row/column of 0s. Because if it's the feature dimension, then this feature is almost useless. If it's the user dimension, it might be possible but are really really rare. So I think a dense array would be good because it is very likely to be **dense** anyway.
Does my reasoning make sense?